api/firmware/psbt_test.go

```diff
@@ -594,3 +594,214 @@ func TestSimulatorBTCPSBTSilentPayment(t *testing.T) {
+
+		input0Pub := simulatorPub(t, device, input0Path...)
+		changePub := simulatorPub(t, device, changePath...)
+		sendSelfPub := simulatorPub(t, device, sendSelfPath...)
+		changeRedeemScript := p2wpkhPkScript(changePub)
+		changePkScript := p2shPkScript(changeRedeemScript)
+		changeAddress, err := btcutil.NewAddressScriptHash(changeRedeemScript, &chaincfg.TestNet3Params)
+		require.NoError(t, err)
+
+		// Previous transaction with one output.
+		prevTx := &wire.MsgTx{
+			Version:  2,
+			LockTime: 0,
+			TxIn: []*wire.TxIn{{
+				PreviousOutPoint: *mustOutpoint("3131313131313131313131313131313131313131313131313131313131313131:0"),
+				Sequence:         0xFFFFFFFF,
+			}},
+			TxOut: []*wire.TxOut{
+				{ // P2TR
+					Value: 100_000_000,
+					PkScript: func() []byte {
+						_, script := makeTaprootOutput(t, input0Pub)
+						return script
+					}(),
+				},
+			},
+		}
+
+		// Spending transaction
+		tx := &wire.MsgTx{
+			Version:  2,
+			LockTime: 0,
+			TxIn: []*wire.TxIn{
+				{PreviousOutPoint: wire.OutPoint{Hash: prevTx.TxHash(), Index: 0}}, // P2TR input
+			},
+			TxOut: []*wire.TxOut{
+				{ // Change output (P2TR)
+					Value:    50_000_000,
+					PkScript: changePkScript,
+				},
+				{ // External output
+					Value:    20_000_000,
+					PkScript: p2wpkhPkScript(sendSelfPub),
+				},
+			},
+		}
+
+		psbt_, err := psbt.NewFromUnsignedTx(tx)
+		require.NoError(t, err)
+
+		// Setup PSBT inputs
+		// Input 0 (P2TR)
+		psbt_.Inputs[0].NonWitnessUtxo = prevTx
+		psbt_.Inputs[0].WitnessUtxo = prevTx.TxOut[0]
+		psbt_.Inputs[0].TaprootInternalKey = schnorr.SerializePubKey(changePub)
+		psbt_.Inputs[0].TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{{
+			XOnlyPubKey:          psbt_.Inputs[0].TaprootInternalKey,
+			MasterKeyFingerprint: binary.LittleEndian.Uint32(fingerprint),
+			Bip32Path:            input0Path,
+		}}
+
+		// Setup change output (P2SH-P2WPKH)
+		psbt_.Outputs[0].Bip32Derivation = []*psbt.Bip32Derivation{{
+			PubKey:               changePub.SerializeCompressed(),
+			MasterKeyFingerprint: binary.LittleEndian.Uint32(fingerprint),
+			Bip32Path:            changePath,
+		}}
+		psbt_.Outputs[0].RedeemScript = changeRedeemScript
+
+		// Setup send-to-self output
+		psbt_.Outputs[1].Bip32Derivation = []*psbt.Bip32Derivation{{
+			PubKey:               sendSelfPub.SerializeCompressed(),
+			MasterKeyFingerprint: binary.LittleEndian.Uint32(fingerprint),
+			Bip32Path:            sendSelfPath,
+		}}
+
+		// Sign & validate
+		require.NoError(t, device.BTCSignPSBT(messages.BTCCoin_TBTC, psbt_, nil))
+		require.NoError(t, psbt.MaybeFinalizeAll(psbt_))
+		require.NoError(t, txValidityCheck(psbt_))
+
+		// v9.22 changed the format of the output string.
+		if device.Version().AtLeast(semver.NewSemVer(9, 22, 0)) {
+			require.Contains(t, stdOut.String(), "ADDRESS: This BitBox (same account): tb1ql34ny8mcpgjqr0ngsnjmlpzjpgncyz2ygh2gye")
+		} else if device.Version().AtLeast(semver.NewSemVer(9, 20, 0)) {
```

Comment:
not for this PR, but I almost feel like we could extract some of these "milestone" versions (9.15.0, 9.20.0, 9.22.0, maybe others) to constants so that they are easier to use and it's less error prone